search/api_test.py

```diff
@@ -32,65 +35,57 @@
 )
 from search.api import (
     get_conn,
-    index_users,
+    recreate_index,
     refresh_index,
-    remove_user,
-    serialize_user,
+    clear_index,
+    index_program_enrolled_users,
+    remove_program_enrolled_user,
+    serialize_program_enrolled_user,
+    USER_DOC_TYPE
 )
 from search.base import ESTestCase
 from search.exceptions import ReindexException
 from search.util import traverse_mapping
+from micromasters.utils import remove_key
 
 
-def remove_key(dictionary, key):
-    """Helper method to remove a key from a dict and return the dict"""
-    del dictionary[key]
-    return dictionary
-
-
-def search():
+class ESTestActions:
     """
-    Execute a search and get results
+    Provides helper functions for tests to communicate with ES
     """
-    # Refresh the index so we can read current data
-    refresh_index()
+    def __init__(self):
+        self.index = settings.ELASTICSEARCH_INDEX
+        self.url = "{}/{}".format(settings.ELASTICSEARCH_URL, self.index)
+        if not self.url.startswith("http"):
+            self.url = "http://{}".format(self.url)
+        self.search_url = "{}/{}".format(self.url, "_search")
+        self.mapping_url = "{}/{}".format(self.url, "_mapping")
 
-    elasticsearch_url = settings.ELASTICSEARCH_URL
-    if not elasticsearch_url.startswith("http"):
-        elasticsearch_url = "http://{}".format(elasticsearch_url)
-    url = urljoin(
-        elasticsearch_url,
-        "{}/{}".format(settings.ELASTICSEARCH_INDEX, "_search")
-    )
-    return get(url).json()['hits']
+    def search(self):
+        """Gets full index data from the _search endpoint"""
```

Comment:
This actually just gets the first 20 or so results, though I'm not sure we run tests with more results than that